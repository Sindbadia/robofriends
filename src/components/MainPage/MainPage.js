import React, { Component } from 'react'
import './MainPage.css'

import Header from '../Header/Header'
import CardList from '../Card/CardList'
import SearchBox from '../SearchBox/SearchBox'
import Scroll from '../Scroll/Scroll'
import ErrorBoundary from '../../hoc/ErrorBoundary'

class MainPage extends Component {
	componentDidMount() {
		this.props.onRequestRobots()
	}

	filterRobots = () => {
		return this.props.robots.filter(robot => {
			return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
		})
	}

	render() {
		return this.props.isPending ? (
			<h1>Loading</h1>
		) : (
			<div className='tc'>
				<Header />
				<SearchBox searchChange={this.props.onSearchChange} />
				<Scroll>
					<ErrorBoundary>
						<CardList robots={this.filterRobots()} />
					</ErrorBoundary>
				</Scroll>
			</div>
		)
	}
}

export default MainPage
