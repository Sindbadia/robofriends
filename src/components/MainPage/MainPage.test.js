import { shallow } from 'enzyme'
import React from 'react'
import { create } from 'react-test-renderer'

import MainPage from './MainPage'

let wrapper
beforeEach(() => {
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: false,
	}
	wrapper = create(<MainPage {...mockProps} />)
})

it('should renders MainPage without crashing', () => {
	expect(wrapper).toMatchSnapshot()
})

it('should filters robots correctly', () => {
	const mockProps2 = {
		onRequestRobots: jest.fn(),
		robots: [
			{
				id: '3',
				name: 'John',
				email: 'john@gmail.com',
			},
		],
		searchField: 'john',
		isPending: false,
	}
	const wrapper2 = shallow(<MainPage {...mockProps2} />)

	// expect(wrapper.instance().filterRobots()).toEqual([])
	expect(wrapper2.instance().filterRobots()).toEqual([
		{
			id: '3',
			name: 'John',
			email: 'john@gmail.com',
		},
	])
})

it('should filters robots correctly 2', () => {
	const mockProps3 = {
		onRequestRobots: jest.fn(),
		robots: [
			{
				id: '3',
				name: 'John',
				email: 'john@gmail.com',
			},
		],
		searchField: 'a',
		isPending: false,
	}
	const filteredRobots = []
	const wrapper3 = shallow(<MainPage {...mockProps3} />)
	// expect(wrapper.instance().filterRobots()).toEqual([])
	expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots)
})
