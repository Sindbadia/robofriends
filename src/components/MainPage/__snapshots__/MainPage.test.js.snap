// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <button
    color="red"
    id="counter"
    onClick={[Function]}
  >
    Count: 
    0
  </button>
  <div
    className="pa2"
  >
    <input
      aria-label="Search Robots"
      className="pa3 ba b--green bg-lightest-blue"
      placeholder="search robots"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "border": "5px solid black",
        "height": "800px",
        "overflowY": "scroll",
      }
    }
  >
    <div />
  </div>
</div>
`;
